SDL2 backend: FingerEvent missing from SDL2 Zig binding

## Description

\`usage/example_sdl2\` and \`usage/example_gui_sdl\` fail to compile because \`input/sdl_input.zig\` references \`sdl.FingerEvent\` which doesn't exist in the SDL2 Zig binding.

## Affected examples

- \`usage/example_sdl2\`
- \`usage/example_gui_sdl\`

## Error

\`\`\`
input/sdl_input.zig:144:46: error: root source file struct 'sdl' has no member named 'FingerEvent'
fn handleFingerEvent(self: *Self, finger: sdl.FingerEvent, phase: TouchPhase) void {
\`\`\`

## Likely cause

The SDL2 Zig wrapper (\`SDL2-0.0.0\`) doesn't expose \`FingerEvent\`. The touch input handler needs to either:
- Use the correct type name from the SDL2 binding (e.g., \`sdl.TouchFingerEvent\`)
- Gate this code behind a platform check (touch only on mobile)
- Update the SDL2 dependency to a version that includes this type

## Reproduction

\`\`\`bash
cd usage/example_sdl2
CI_TEST=1 zig build run
\`\`\`

--- a/input/sdl_input.zig
+++ b/input/sdl_input.zig
@@ -141,7 +141,7 @@ fn processEvent(self: *Self, event: sdl.Event) void {
 }
 
 /// Handle SDL finger/touch events
-fn handleFingerEvent(self: *Self, finger: sdl.FingerEvent, phase: TouchPhase) void {
+fn handleFingerEvent(self: *Self, finger: sdl.TouchFingerEvent, phase: TouchPhase) void {
     // SDL finger coordinates are normalized (0-1), convert to screen coords
     // Note: SDL requires window dimensions for proper conversion, for now use raw coords
     const touch = Touch{

